app/models/secret_storage.rb

```diff
@@ -50,6 +50,49 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    VAULT_SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(gen_vault_path(key))
+      return false if result.nil?
+      result.data[:secret_data]
+    end
+
+    def self.write(key, data)
+      Vault.logical.write(gen_vault_path(key), secret_data: data)
+    end
+
+    def self.delete(key)
+      Vault.logical.delete(gen_vault_path(key))
+    end
+
+    def self.keys()
+      Vault.logical.list(VAULT_SECRET_BACKEND).map { |key| encode_path(key) }
```

Comment:
needs it to list all the keys at the root of the storage engine.  nav works as intended.